dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/setup/ExecutorServiceBuilder.java

```diff
@@ -116,6 +119,36 @@ public ExecutorService build() {
         return executor;
     }
 
+    static String getNameWithoutFormat(String nameFormat) {
```

Comment:
I'm sure there is a better way to determine common prefix + suffix of two strings. This basically formats the string with 0 and then strips off the common prefix/suffix. If it finds a trailing '-' in the common prefix it removes it from the string.

This will take the common pattern 'jersey-client-somename-%d' and return 'jersey-client-somename'.

To get more flexibility, we could change ExecutorServiceBuilder to take a name (without a format string) to be used as the prefix for metrics.